homeassistant/components/ecovacs/config_flow.py

```
@@ -0,0 +1,108 @@
+        _LOGGER.exception("Unexpected exception")
+        errors["base"] = "unknown"
+
+    return errors
+
+
+class EcovacsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ecovacs."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            errors = await self.hass.async_add_executor_job(validate_input, user_input)
+
+            if not errors:
+                self._async_abort_entries_match(
+                    {CONF_USERNAME: user_input[CONF_USERNAME]}
+                )
+                return self.async_create_entry(
+                    title=user_input[CONF_USERNAME], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_USERNAME): selector.TextSelector(
+                            selector.TextSelectorConfig(
+                                type=selector.TextSelectorType.TEXT
+                            )
+                        ),
+                        vol.Required(CONF_PASSWORD): selector.TextSelector(
+                            selector.TextSelectorConfig(
+                                type=selector.TextSelectorType.PASSWORD
+                            )
+                        ),
+                        vol.Required(CONF_COUNTRY): vol.All(vol.Lower, cv.string),
+                        vol.Required(CONF_CONTINENT): vol.All(vol.Lower, cv.string),
+                    }
+                ),
+                user_input,
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, user_input: dict[str, Any]) -> FlowResult:
+        """Import configuration from yaml."""
+        async_create_issue(
```

We should raise different issues if we can't connect and invalid Auth